flang/include/flang/Optimizer/Dialect/FIRTypes.td

```diff
@@ -224,6 +224,28 @@ def fir_LogicalType : FIR_Type<"Logical", "logical"> {
   }];
 }
 
+def fir_LLVMPointerType : FIR_Type<"LLVMPointer", "llvm_ptr"> {
```

Comment:
I do not think they are precedents. I tend to agree with your point but I am not sure about the best name here. I'll keep it like this because that is what we had discussed with Eric. I will do a NFC change later if we agree on a better name.

Reply:
`llvm_ptr` is the right name. It will map exactly to `!llvm.ptr` in the LLVM-IR dialect and that maps exactly to LLVM's pointer type.